### ISSUE TITLE
Can't start a radio after updating station profile

### ISSUE THREAD

**Issue by hecgua** [AzuraCast/AzuraCast#5235]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Stable Channel

### Current AzuraCast Version

v0.15.2 Stable

### What happened?

After banning a user agent (Roku) on a particular station profile that was making more than 12 simultaneous connections on Icecast I couldn't restart the station. Tried everything to restart the stations, just the Icecast instance restarted, the hiss sound was listening but no songs were heard. Tried many times to hit the "Restart to Apply Changes", "restart and "stop/start" LS but nothing seems to work. Also tried to Rebuild AutoDJ Queue and didn't work.

FYI: I don't have any special configuration of any kind on my radios.

Some times when trying to restart, I got this Azuracast banner: NotRunningException at /var/azuracast/www/src/Radio/AbstractAdapter.php L269: Liquidsoap cannot stop
It is not running.

I did this banning procedure with other 8 stations to prevent this connection abuse and had no problem with them.

### Relevant log output

```shell
2022/03/25 23:51:06 >>> LOG START
2022/03/25 23:51:00 [main:3] Liquidsoap 2.0.3-1
2022/03/25 23:51:00 [main:3] Using: bytes=[distributed with OCaml 4.02 or above] posix-time2=2.0.0 pcre=7.5.0 sedlex=2.5 menhirLib=20211128 curl=0.9.2 memtrace=v0.2.1.2 mem_usage=0.0.1 dtools=0.4.4 duppy=0.9.2 cry=0.6.6 mm=0.8.0 xmlplaylist=0.1.5 lastfm=0.3.3 ogg=0.7.0 ogg.decoder=0.7.0 vorbis=0.8.0 vorbis.decoder=0.8.0 opus=0.2.1 opus.decoder=0.2.1 speex=0.4.0 speex.decoder=0.4.0 mad=0.5.0 flac=0.3.0 flac.ogg=0.3.0 flac.decoder=0.3.0 dynlink=[distributed with Ocaml] lame=0.3.5 shine=0.2.2 frei0r=0.1.2 fdkaac=0.3.2 theora=0.4.0 theora.decoder=0.4.0 ffmpeg=1.1.1 bjack=0.1.6 alsa=0.3.0 ao=0.2.3 samplerate=0.1.6 taglib=0.3.9 ssl=0.5.9 magic=0.7.3 camomile=1.0.2 inotify=2.3 yojson=1.7.0 faad=0.5.0 soundtouch=0.1.9 portaudio=0.2.3 pulseaudio=0.1.4 ladspa=0.2.0 dssi=0.1.3 tsdl=v0.9.8 tsdl-ttf=0.3.2 tsdl-image=0.3.2 camlimages=4.2.6 cohttp-lwt-unix=5.0.0 prometheus-app=1.1 srt.constants=0.2.2 srt.types=0.2.2 srt.stubs=0.2.2 srt.stubs.locked=0.2.2 srt=0.2.2 lo=0.2.0 gd=1.0a5
2022/03/25 23:51:02 [clock:3] Using native (high-precision) implementation for latency control
2022/03/25 23:51:06 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:51:06 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:51:06 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:51:06 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:51:06 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:51:06 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:51:06 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:51:06 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:51:06 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:51:06 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:51:06 [frame:3] Using 44100Hz audio, 25Hz video, 44100Hz main.
2022/03/25 23:51:06 [frame:3] Video frame size set to: 1280x720
2022/03/25 23:51:06 [frame:3] Frame size must be a multiple of 1764 ticks = 1764 audio samples = 1 video samples.
2022/03/25 23:51:06 [frame:3] Targeting 'frame.duration': 0.04s = 1764 audio samples = 1764 ticks.
2022/03/25 23:51:06 [frame:3] Frames last 0.04s = 1764 audio samples = 1 video samples = 1764 ticks.
2022/03/25 23:51:06 [sandbox:3] Running inside a docker container, disabling sandboxing.
2022/03/25 23:51:06 [video.converter:3] Using preferred video converter: ffmpeg.
2022/03/25 23:51:06 [audio.converter:3] Using samplerate converter: ffmpeg.
2022/03/25 23:51:06 [error_jingle:3] /usr/local/share/icecast/web/error.mp3 is static, resolving once for all...
2022/03/25 23:52:04 [main:3] Shutdown started!
2022/03/25 23:52:04 [threads:3] Shutting down scheduler...
2022/03/25 23:52:04 [threads:3] Scheduler shut down.
2022/03/25 23:52:04 [main:3] Cleaning downloaded files...
2022/03/25 23:52:04 [main:3] Freeing memory...
2022/03/25 23:52:11 >>> LOG START
2022/03/25 23:52:06 [main:3] Liquidsoap 2.0.3-1
2022/03/25 23:52:06 [main:3] Using: bytes=[distributed with OCaml 4.02 or above] posix-time2=2.0.0 pcre=7.5.0 sedlex=2.5 menhirLib=20211128 curl=0.9.2 memtrace=v0.2.1.2 mem_usage=0.0.1 dtools=0.4.4 duppy=0.9.2 cry=0.6.6 mm=0.8.0 xmlplaylist=0.1.5 lastfm=0.3.3 ogg=0.7.0 ogg.decoder=0.7.0 vorbis=0.8.0 vorbis.decoder=0.8.0 opus=0.2.1 opus.decoder=0.2.1 speex=0.4.0 speex.decoder=0.4.0 mad=0.5.0 flac=0.3.0 flac.ogg=0.3.0 flac.decoder=0.3.0 dynlink=[distributed with Ocaml] lame=0.3.5 shine=0.2.2 frei0r=0.1.2 fdkaac=0.3.2 theora=0.4.0 theora.decoder=0.4.0 ffmpeg=1.1.1 bjack=0.1.6 alsa=0.3.0 ao=0.2.3 samplerate=0.1.6 taglib=0.3.9 ssl=0.5.9 magic=0.7.3 camomile=1.0.2 inotify=2.3 yojson=1.7.0 faad=0.5.0 soundtouch=0.1.9 portaudio=0.2.3 pulseaudio=0.1.4 ladspa=0.2.0 dssi=0.1.3 tsdl=v0.9.8 tsdl-ttf=0.3.2 tsdl-image=0.3.2 camlimages=4.2.6 cohttp-lwt-unix=5.0.0 prometheus-app=1.1 srt.constants=0.2.2 srt.types=0.2.2 srt.stubs=0.2.2 srt.stubs.locked=0.2.2 srt=0.2.2 lo=0.2.0 gd=1.0a5
2022/03/25 23:52:08 [clock:3] Using native (high-precision) implementation for latency control
2022/03/25 23:52:11 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:52:11 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:52:11 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:52:11 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:52:11 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:52:11 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:52:11 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:52:11 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:52:11 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:52:11 [playlist.basic:3] Wrong mime type application/csv for playlist!
2022/03/25 23:52:11 [frame:3] Using 44100Hz audio, 25Hz video, 44100Hz main.
2022/03/25 23:52:11 [frame:3] Video frame size set to: 1280x720
2022/03/25 23:52:11 [frame:3] Frame size must be a multiple of 1764 ticks = 1764 audio samples = 1 video samples.
2022/03/25 23:52:11 [frame:3] Targeting 'frame.duration': 0.04s = 1764 audio samples = 1764 ticks.
2022/03/25 23:52:11 [frame:3] Frames last 0.04s = 1764 audio samples = 1 video samples = 1764 ticks.
2022/03/25 23:52:11 [sandbox:3] Running inside a docker container, disabling sandboxing.
2022/03/25 23:52:11 [video.converter:3] Using preferred video converter: ffmpeg.
2022/03/25 23:52:11 [audio.converter:3] Using samplerate converter: ffmpeg.
2022/03/25 23:52:12 [error_jingle:3] /usr/local/share/icecast/web/error.mp3 is static, resolving once for all...
```

**Comment by SlvrEagle23** [comment]
@hecgua The logs there don't seem to indicate any real problem that would suggest why it's actually crashing...does restarting the entire installation help at all?

**Comment by hecgua** [comment]
@SlvrEagle23 I didn't try to restart the entire Azuracast instance but I finally managed to restart the station.

What worked for me, after trying many things, was to stop LS, then disable all playlists (about 8) except the "default" playlist. After this, I started LS and it finally played audio. After listening to a few songs from the default playlist, I enabled the rest of the playlists again and after several more songs, everything seems to be working fine.

**Comment by SlvrEagle23** [comment]
@hecgua That's...certainly interesting. I have no idea why that would've fixed things, but I'm glad you were able to find a solution in any case. We're continuing to investigate stability issues overall with the latest Liquidsoap, and will keep publishing updated versions of our software to incorporate those fixes as they come in.

**Comment by hecgua** [comment]
@SlvrEagle23 

About this report that I made a few days ago, it turned out that this action of restarting the radio with a single playlist, has kept the LS instance with a low CPU consumption since then, relatively normal and without sudden increases. Even enabling the rest of the playlists again after the restart.

The same day of the report I did this same action (to restart the radio with a single playlist) with some of my other radios and finally, in several days, their respective instances of LS have been very stable. Today I verified that this action is helping a lot to keep the LS instances much more stable and with a very even CPU load, since today, one of the radios that had not been restarted with a single playlist, finally failed. After doing this restart procedure, the radio has been very stable.

In the next few days I will be observing the behavior of my installation in terms of the consumption of the LS instances and, above all, the behavior of mysqld to see if these procedures stop triggering the load of the rest of the radios and other instances. 

For my way of seeing things, the failures that have occurred lately in Azuracast have some relation to the start of operations of each radio and the processing of the playlists in LS at the beginning.

Hope this information will be of some use to help the work of the Azuracast collaborators.
